lab_2/main.py

```diff
@@ -13,7 +13,13 @@ def get_freq_dict(tokens: list) -> dict or None:
     :param tokens: a list of tokens
     :return: a dictionary with frequencies
     """
-    pass
+    if not isinstance(tokens, list) or None in tokens:
+        return None
+    frequency_dictionary = {}
+    for word in tokens:
+        if isinstance(word, str):
+            frequency_dictionary[word] = round(tokens.count(word)/len(tokens), 5)
```

Comment:
Are you sure about calculating the length each time?

Reply:
i don't know, at least it works.....